src/Nethermind/Nethermind.Network/P2P/Subprotocols/Etha/EthaProtocolHandler.cs

```diff
@@ -0,0 +1,82 @@
+using Nethermind.Blockchain;
+using Nethermind.Core;
+using Nethermind.Logging;
+using Nethermind.Network.P2P.Subprotocols.Etha.Messages;
+using Nethermind.Stats;
+using System.Linq;
+
+namespace Nethermind.Network.P2P.Subprotocols.Etha
+{
+    public class EthaProtocolHandler : ProtocolHandlerBase
+    {
+        private readonly IBlockTree _blockTree;
+        private readonly ILogger _logger;
+
+        public EthaProtocolHandler(
+            IBlockTree blockTree,
+            IMessageSerializationService serializer,
+            INodeStatsManager nodeStats,
+            ILogManager logManager) 
+            : base(serializer, nodeStats, logManager)
+        {
+            _blockTree = blockTree;
+            _logger = logManager.GetClassLogger();
+
+            // Register message serializers
+            serializer.Register(new GetShardedBlocksMessageSerializer());
+            serializer.Register(new ShardedBlocksMessageSerializer());
+            serializer.Register(new NewShardedBlockMessageSerializer());
+        }
+
+        public override void HandleMessage(Packet packet)
+        {
+            switch(packet.PacketType)
+            {
+                case EthaMessageCode.GetShardedBlocks:
+                    Handle(Deserialize<GetShardedBlocksMessage>(packet.Data));
+                    break;
```

Comment:
Would be good to add metrics to all messages. Compare please with implementations of other protocols. But it's just good-to-have, not hard requirement - we can add it later, in another PR

Reply:
e.g. `GetShardedBlocksReceived` and here `Metrics.GetShardedBlocksReceived++`